### ISSUE TITLE
Bug in MimeParser::GetMultiLine

### ISSUE THREAD

**Issue by kleuter** [eneam/mboxviewer#12]
Hi,

Thunderbird displays this email with "Jack Daniels" in the subject line:

```
From: sales@hidden.com
Reply-To: sales@hidden.com
To: hidden@gmail.com
Message-ID: <10427239.616391172412534598.JavaMail.root@app1sc1.hidden.com>
Subject: New Order for 1 license of Uninstall Tool-Full Version,  Jack
 Daniels (Ref. # 16558187)
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Transfer-Encoding: 7bit
```
mboxviewer displays "JackDaniels" (no space).

I think that nextLine.TrimLeft() shouldn't be called here. Actually we should trim a SINGLE space when the line begins with TWO spaces .

Thanks for the app anyway, it does a nice job for me.

**Comment by zigm** [comment]
Thanks for reporting the issue. I seem to remember I had issues when
discarding all spaces or not discarding any or  discarding all except one.
I will investigate and provide an update.

Thank you,
Zbigniew

**Comment by zigm** [comment]
According to RFC documents, unfolding is simple, just remove CRLF to unfold.  Current implementation discards CRLF and all white spaces. I believe RFC recommended approach was breaking words/token by inserting white spaces in a few cases. Maybe these emails didn't do folding correctly, not sure now. Not sure how folding can work if there is no white spaces in the body of the header field but such case should be rather rare I think.

Your rules is the same what I was thinking in the past and it would cover both cases, i.e folding could happen anywhere in the body of the header field by inserting CRLF and one extra space that could be removed during the unfolding. I don't see RFC saying that.  Still it might be best to follow the RFC (assuming I am reading the RFC 822 and 5322 correctly). It may not work for some emails possibly not formatted correctly.  I am planning to examine possible solutions against mail archives I have.

Will provide an update.

Thank you,
Zbigniew

**Comment by zigm** [comment]
Released v1.0.3.16 of the mbox viewer to resolve the reported issue. I basically commented out the original line that replaces CRLF and white spaces with the single blank character.  Could not find old emails which I thought the change was braking.
Other changes are listed in the README file. Thanks again for reporting the issue.

Thank You
Zbigniew
